js/interface-lists.js

```diff
@@ -509,12 +529,48 @@ function attahObservers() {
       });
   });
 
+  function highlightError(fieldIds) {
+      _.each(fieldIds, function(id) {
+        $('[data-field-id="' + id + '"] .text-danger').removeClass('hidden');
+      });
+    }
+
+    function checkImageFoldersSelection() {
+      if (!widgetData['summary-fields']) {
+        highlightError(selectedFoldersId);
+        return false;
+      }
+
+      var totalArray = _.concat(widgetData.detailViewOptions, widgetData['summary-fields']);
+      var errorInputIds = _.filter(selectedFoldersId, function(id) {
+          return !_.some(totalArray, function(item) {
+            return item.id === id && item.folder;
+          });
+      });
+
+      highlightError(errorInputIds);
+      return errorInputIds.length === 0;
+    }
+
   Fliplet.Widget.onSaveRequest(function () {
     if (!dynamicLists.isLoaded) {
       Fliplet.Widget.complete();
       return;
     }
-    $('form').submit();
+    var isImageFoldersSelected = checkImageFoldersSelection();
+
+    if (isImageFoldersSelected || filePickerPromises.length) {
+      $('.text-danger').addClass('hidden');
+      $('form').submit();
+      return;
+    }
+
+    if (!isImageFoldersSelected) {
```

Comment:
Move this block above so that validation clauses are executed first, and the use `return` to stop further code execution.

Reply:
Cannot do it because then it will not allow you to select a folder from file-picker.